core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1475,6 +1486,10 @@ private Optional<Throwable> acknowledgeCompleteBatch(
         return Optional.empty();
     }
 
+    LogOffsetMetadata latestFetchOffsetMetadata() {
```

Comment:
Is it need to be default scoped?

Reply:
I've moved it to protected. It cannot be private since it is utilized in `DelayedShareFetch`